src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -72,6 +100,63 @@ export const selectTotalStakingDelegationCryptoByAccountSpecifier = createSelect
   }
 )
 
+export const selectTotalStakingDelegationCryptoByFilter = createSelector(
+  selectStakingDataByFilter,
+  selectAssetIdParamFromFilterOptional,
+  selectAccountIdParamFromFilterOptional,
+  (state: ReduxState) => state.assets.byId,
+  // We make the assumption that all delegation rewards come from a single denom (asset)
+  // In the future there may be chains that support rewards in multiple denoms and this will need to be parsed differently
+  (stakingData, assetId, _, assets) => {
+    const amount = reduce(
+      stakingData?.delegations,
+      (acc, delegation) => acc.plus(bnOrZero(delegation.amount)),
+      bn(0)
+    )
+
+    return fromBaseUnit(amount, assets[assetId].precision ?? 0).toString()
+  }
+)
+
+export const selectAllStakingDelegationCrypto = createSelector(selectStakingData, stakingData => {
+  const allStakingData = Object.entries(stakingData.byAccountSpecifier)
+  return reduce(
+    allStakingData,
+    (acc, val) => {
```

Comment:
You should destructure this here `[accountId, delegationObj]` so you don't have to use `val[0]`